otumat/usage.py

```diff
@@ -0,0 +1,528 @@
+"""Library for package usage data management."""
+from pathlib import Path
+from json import load, dump, dumps, loads
+from os import makedirs, getenv, system
+from flask import Flask, request
+from appdirs import user_data_dir
+from shutil import rmtree
+from datetime import datetime
+import webbrowser
+from logging import getLogger, ERROR as log_error
+from threading import Thread
+# client info
+from re import findall
+from uuid import getnode
+from sys import version_info, stdin, platform as operating_system
+from platform import system as general_system
+from subprocess import Popen, PIPE, DEVNULL
+from pkg_resources import get_distribution
+from contextlib import closing
+from socket import socket, AF_INET, SOCK_DGRAM, SOCK_STREAM, SOL_SOCKET, SO_REUSEADDR
+from urllib.request import urlopen
+from urllib.parse import urlencode
+from time import tzname
+# logging
+from sqlite3 import connect
+# sending
+from urllib import request as urllib_request
+from urllib.error import HTTPError, URLError
+from base64 import b64encode
+from time import sleep
+
+
+class UsageAgent:
+    """
+    Local agent which buffers package usage data locally and periodically sends logged data to
+    an OAuth2-compatible endpoint.
+    """
+    def __init__(self, author: str, data_directory: str, package_name: str, host: str = None,
```

Comment:
Force keyword arguments for future proofing.
```suggestion
    def __init__(self, *, author: str, data_directory: str, package_name: str, host: str = None,
```

Reply:
Good point, I've updated all methods to require kwargs now.